mongodb/connection_test.go

```diff
@@ -134,7 +134,7 @@ func TestSuite(t *testing.T) {
 			Convey("UpsertId should insert if not exists", func() {
 				_, err := conn.
 					Collection(collection).
-					UpsertById(context.Background(), 4, bson.M{"$set": bson.M{"new_key": 456}})
+					UpsertOne(context.Background(), bson.M{"_id": 4}, bson.M{"$set": bson.M{"new_key": 456}})
```

Comment:
The convey descriptions refer to `UpsertId`, can we amend them so they match what they are testing?

Also, are we removing test coverage for `UpsertById`? Even if it is deprecated I think we should keep coverage? 

I am not 100% convinced of deprecating it anyway. I see it as a valid convenient function to avoid callers having to create a `bson.M`?

Reply:
Yeah wasn't entirely sure of deprecating the xxxBy Id myself, but I thought on balance it was such a small convenience that it might even obscure that it was updating one and only one?
On the test coverage yes I forgot to tidy that up